cmd/local-gadget/trace/trace.go

```diff
@@ -15,13 +15,91 @@
 package trace
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cilium/ebpf"
 	"github.com/spf13/cobra"
 
 	commontrace "github.com/kinvolk/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/kinvolk/inspektor-gadget/cmd/common/utils"
+	"github.com/kinvolk/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadget-collection/gadgets/trace"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	localgadgetmanager "github.com/kinvolk/inspektor-gadget/pkg/local-gadget-manager"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
 )
 
+// TraceGadget represents a gadget belonging to the trace category.
+type TraceGadget[Event commontrace.TraceEvent] struct {
+	commonFlags *utils.CommonFlags
+	parser      commontrace.TraceParser[Event]
+	runTracer   func(*ebpf.Map, gadgets.DataEnricher, func(Event)) (trace.Tracer, error)
```

Comment:
IMO, the confusion comes from the fact that `NewTracer` create and also starts the tracer 😅

Reply:
Maybe `NewTracer` can also be renamed? Maybe `RunTracer`? 

I mean, it is strange that now we have `createAndRunTracer` with a code that only calls `NewTracer`. It makes reader think that code is incomplete and it is not really running the tracer but just creating it, don't you think?

```golang
	createAndRunTracer: func(mountnsmap *ebpf.Map, enricher gadgets.DataEnricher, eventCallback func(types.Event)) (trace.Tracer, error) {
		config := &tracer.Config{
			MountnsMap:   mountnsmap,
			TargetPid:    flags.TargetPid,
			TargetPorts:  flags.ValidatedTargetPorts,
			IgnoreErrors: flags.IgnoreErrors,
		}

		return tracer.NewTracer(config, enricher, eventCallback)
	},
```

Of course, we can mange it in another PR because such change implies also updating the examples and all the gadget-collectors ...

Reply:
https://github.com/kinvolk/inspektor-gadget/issues/905